torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3342,11 +3342,20 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::_allgather_base(
           c10::cuda::CUDACachingAllocator::recordStream(
               output.storage().data_ptr(), stream);
         }
+
+        ncclDataType_t nccl_dtype;
+        if (input.scalar_type() == at::kFloat8_e5m2 ||
+            input.scalar_type() == at::kFloat8_e4m3fn) {
+          nccl_dtype = ncclInt8;
+        } else {
+          nccl_dtype = getNcclDataType(input.scalar_type());
```

Comment:
just curious, why not add the type mapping into `ncclDataType` above so that `getNcclDataType` returns it?

Are we wanting to potentially have different behavior for the type depending on which collective we're in?

Reply:
Yes this is based on feedback from someone in the PyTorch team (I think it was in workplace chat).

Reply:
ping @kwen2501  @H-Huang to comment further- do we want it this way? or inside ncclDataType